applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -797,6 +798,11 @@ static void discover_sink_cb(struct bt_conn *conn, struct bt_codec *codec, struc
 			for (int i = 0; i < codec->meta_count; i++) {
 				store_loc->meta[i].data.data = store_loc->meta[i].value;
 			}
+
+			/* In temp_cap.num_caps we want to store the actual number
+			 * of capabilities, while params->num_caps is an iterator
+			 */
+			temp_cap[temp_cap_index].num_caps = params->num_caps + 1;
```

Comment:
Should there be a check if num_caps exceeds CONFIG_CODEC_CAP_COUNT_MAX?

Reply:
Here and on line 933

Reply:
params->num_caps is inderectly cheched against CONFIG_CODEC_CAP_COUNT_MAX on line 788. temp_cap[temp_cap_index].codec has the length of CONFIG_CODEC_CAP_COUNT_MAX